echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/WebhookEventHandler.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.scm;
+
+import com.netflix.spinnaker.echo.model.Event;
+import java.util.Map;
+
+public interface WebhookEventHandler {
+  boolean shouldSendEvent(Event event);
+  boolean handles(String source);
+  void handle(Event event, Map postedEvent);
```

Comment:
handle is responsible for parsing the event to pull out branch, repoProject, slug, and hash but i wonder if it makes more sense to pull each of these out into individual getters? thoughts?

Reply:
I would do it more explicit:
1. Rename `WebhookEventHandler` to `GitWebhookEventHandler`
2. Have one method `extractContent` which maybe returns `GitEvent.Content`
  Not so glad with the _content_ name.
3. Have another method `generateMap` which maps `GitEvent.Content` to `Map`
4. Put that all together by calling `event.putAll(generatedMap)`

Reply:
I'd also like to encourage class & method docs on interfaces. No one should have to hunt down this PR to get context as to the intended functionality of particular methods.

Reply:
Documentation added to `GitWebhookHandler`!